pkg/usage/metrics.go

```diff
@@ -0,0 +1,208 @@
+) error {
+	klog.Info("usage.Metrics: Count console users...\n")
+	startTime := time.Now()
+
+	ctx := context.TODO()
+	config := &rest.Config{
+		Host:        userSettingsEndpoint,
+		BearerToken: serviceAccountToken,
+		Transport:   userSettingsClient.Transport,
+	}
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		klog.Errorf("usage.Metrics: Failed to create service account client: %v\n", err)
+		return err
+	}
+
+	roleBindingList, err := client.RbacV1().RoleBindings("openshift-console-user-settings").List(ctx, metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("usage.Metrics: Failed to get user-settings RoleBindings: %v\n", err)
+		return err
+	}
+
+	canGetNamespaceAccessReview := &authv1.SelfSubjectAccessReview{
+		Spec: authv1.SelfSubjectAccessReviewSpec{
+			ResourceAttributes: &authv1.ResourceAttributes{
+				Verb:     "get",
+				Resource: "namespaces",
+			},
+		},
+	}
+
+	var clusterAdmins, developers, unknowns float64
+
+	for _, roleBinding := range roleBindingList.Items {
+		if !strings.HasPrefix(roleBinding.Name, "user-settings-") || !strings.HasSuffix(roleBinding.Name, "-rolebinding") {
+			klog.Infof("usage.Metrics: Ignore role binding: %s (name doesn't match user-settings-*-rolebinding)\n", roleBinding.Name)
+			continue
+		}
+		if len(roleBinding.Subjects) != 1 {
+			klog.Infof("usage.Metrics: Ignore role binding: %s (unexpected subject length %v != 1)\n", roleBinding.Name, len(roleBinding.Subjects))
+			unknowns++
+			continue
+		}
+		user := roleBinding.Subjects[0]
+		if user.Kind != "User" {
+			klog.Infof("usage.Metrics: Ignore role binding: %s (unexpected subject kind %v != 'User')\n", roleBinding.Name, user.Kind)
+			unknowns++
+			continue
+		}
+
+		impersonatedConfig := &rest.Config{
+			Host:        userSettingsEndpoint,
+			BearerToken: serviceAccountToken,
+			Transport:   userSettingsClient.Transport,
+			Impersonate: rest.ImpersonationConfig{
+				UserName: user.Name,
+			},
+		}
+		impersonatedClient, err := kubernetes.NewForConfig(impersonatedConfig)
+		if err != nil {
+			klog.Errorf("usage.Metrics: Error when creating client to check permissions for %v: %v\n", user.Name, err)
+			unknowns++
+			continue
+		}
+
+		res, err := impersonatedClient.AuthorizationV1().SelfSubjectAccessReviews().Create(
+			ctx,
+			canGetNamespaceAccessReview,
+			metav1.CreateOptions{},
+		)
+		if err != nil {
+			klog.Errorf("usage.Metrics: Error when checking permissions for %v: %v\n", user.Name, err)
+			unknowns++
+			continue
+		}
+
+		if res.Status.Allowed {
+			if clusterAdmins < 10 {
+				klog.Infof("usage.Metrics: Count %v as cluster-admin...\n", user.Name)
+			}
+			clusterAdmins++
+		} else {
+			if developers < 10 {
+				klog.Infof("usage.Metrics: Count %v as developer...\n", user.Name)
+			}
+			developers++
+		}
+	}
+
+	klog.Infof("usage.Metrics: Update users metrics: %v cluster-admins, %v developer, %v unknowns/errors (tooked %v)\n", clusterAdmins, developers, unknowns, time.Since(startTime))
```

Comment:
Usually the names are quoted
```suggestion
	klog.Infof("usage.Metrics: Update users metrics: %q cluster-admins, %q developer, %v unknowns/errors (took %v)\n", clusterAdmins, developers, unknowns, time.Since(startTime))
```